feat: add grype and syft from anchore team (#895)
tektoncd/catalog: task/syft/0.1/README.md

# syft

> This task allows the use of the syft for Tekton Pipelines.See <https://github.com/anchore/syft>

Author: thanks

## What's Syft?

A CLI tool and Go library for generating a Software Bill of Materials (SBOM) from container images and filesystems. Exceptional for vulnerability detection when used with a scanner tool like Grype.
For more detail, please [see](https://github.com/anchore/syft).

## Parameters

> * **SYFT_IMAGE**: Optional address of the grype container image to be used for task.

Reviewer: ```suggestion
* **SYFT_IMAGE**: Optional address of the syft container image to be used for task.
```

Author: good catch, thanks.

  _default_: "docker.io/anchore/syft:v0.35.1@sha256:fd2da1424585680f220ed61db13096f7abcd0c0073b52616bbce397a8e708a96"

> * **ARGS**: The arguments to pass to `grype` CLI. This parameter is required to run this task.

Reviewer: ```suggestion
* **ARGS**: The arguments to pass to `syft` CLI. This parameter is required to run this task.
```

Author: good catch, thanks.

### Examples

> Run `syft --help` for Grype usage.

Reviewer: ```suggestion
Run `syft --help` for Syft usage.
```

Author: thanks

Using the [Tekton CLI](https://github.com/tektoncd/cli/blob/main/docs/cmd/tkn_task_start.md) (`tkn`):

```shell

> tkn task start syft -p ARGS="--help" --workspace name=source-dir,emptyDir=""

Reviewer: Is the `--workspace` part required since it's optional in the task?

Author: IMHO, it must not be an optional thing, so, I removed `optional` parameter, thanks

```

> Scan Alpine Image:

Author: thanks

```shell
tkn task start syft -p ARGS="docker.io/alpine:3.13" --workspace name=source-dir,emptyDir=""
```

## Platforms

> The Task can be run on `linux/amd64` platform.

Reviewer: ```suggestion
The Task can be run on `linux/amd64` and `linux/arm64` platforms.
```

```
$ crane config anchore/syft:v0.35.1 | jq .architecture
"amd64"

$ crane config anchore/syft:v0.35.1-arm64v8 | jq .architecture
"arm64"
```

Author: thanks
